PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/spark/ui/filesystem/AdlsGen2VirtualFile.java

```diff
@@ -0,0 +1,71 @@
+package com.microsoft.azure.hdinsight.spark.ui.filesystem;
+
+import com.intellij.openapi.vfs.VirtualFile;
+import com.intellij.openapi.vfs.VirtualFileSystem;
+import com.intellij.openapi.vfs.impl.http.HttpFileSystemImpl;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class AdlsGen2VirtualFile extends AzureStorageVirtualFile {
+    private String name;
+    private String path;
+    private VirtualFileSystem fileSystem;
+    private boolean isDirectory;
+    private String prefix;
+    private List<VirtualFile> children = new ArrayList<>();
+    private VirtualFile parent;
```

Comment:
parent cannot be null otherwise intelliJ will throw exception

---
In reply to: [285420957](https://github.com/microsoft/azure-tools-for-java/pull/3124#discussion_r285420957) [](ancestors = 285420957)